components/speedreader/rust/lib/src/readability/src/scorer.rs

```diff
@@ -129,43 +144,53 @@ pub fn fix_img_path(data: &ElementData, url: &Url) -> Option<Url> {
     }
 }
 
+bitflags::bitflags! {
+    pub struct ImageLoadedMask: u8 {
+        const SRC = 1 << 0;
+        const SRCSET = 1 << 1;
+
+        const NONE = 0;
+    }
+}
 /// Returns true if the image src loads without JavaScript logic. Some sites like Kotaku and The
 /// Atlantic try and get fancy with this. Our simple heuristic is to check if src is set something
 /// reasonable or srcset is set at all. Some things we've seen in the wild are srcset being left
 /// out in favor of data-srcset, and src being base64 encoded.
 #[inline]
-fn img_is_loaded(data: &ElementData) -> bool {
-    let mut loaded = false;
+fn img_is_loaded(data: &ElementData) -> ImageLoadedMask {
+    let mut mask: ImageLoadedMask = ImageLoadedMask::NONE;
```

Comment:
I can't figure out why there's a mask and BitOr in this context to figure out whether the img has been loaded

Reply:
So an img can be loaded with a `src` or a `srcset`. If the img has neither, it gets deleted. it will always try and load both the `src` and the `srcset` if it is lazy loaded, because often times the `srcset` is higher quality. For example, if you have Firefox and go to [this page](https://restofworld.org/2021/the-rise-and-fall-of-missed-calls-in-india/), then activate reader mode, you will see that some of the photos are tiny. This is because they didn't try and lazy load the `srcset`.

Reply:
Gotcha, makes sense. Thanks!